apps/data-types.md

```diff
@@ -4,111 +4,145 @@ key: apps-data-types
 title: Data types
 ---
 
-In the following the JSON supported data types are listed.
+# address
 
-# Number
+This object is used for the attributes of shippingAddress and billingAddress.
+
+| Attribute | Type | Description |
+| - | :-: |  - |
+| firstName | string | The first name of the customer or supplied party.|
+| lastName | string | The last name of the customer or supplied party.|
+| street | string | The street name of the address. |
+| zipCode | string | The zip or postal code of the address. |
+| city | string | The city of the address. |
+| country | string | The country of the address. |
+| birthday | string | The date of birth of the customer. |
+| emailAddress | string | The email address of the customer. |
 
-An integer or floating point.
+# attributeSelection
 
-| Type      | Description    |
-|---------------|---------------|
-| Integer   | Digits 1-9, 0 and positive or negative |
-| Floating point | Digit string of any length with a radix point, e.g. 15289.1104|
+| Attribute | Type | Description |
+| - | :-: |  - |
+| name | string | The name of the selected product attribute, e.g. colour. |
+| value | string | The assigned value of the selected product attribute, e.g. white. |
 
-## Example
+# customAttribute
 
-{% highlight text %}
-{"page": 1}
-{% endhighlight %}
+| Attribute | Type | Description |
+| - | :-: |  - |
+| key | string | The identifier of the custom attribute. |
+| displayKey | string | The displayed name of the custom attribute. |
+| singleValue | boolean | `true` if just one feature is selected for the custom attribute.  |
+| type | string | The data type of the custom attribute. |
+| values | array of [value](page:apps-data-types#value)| The options selected for the custom attribute. |
 
-# String
+# image
 
-A sequence of characters.
+This object is used for the attributes of images.
 
-## Example
+| Attribute | Type | Description |
+| - | :-: |  - |
+| url | string | The URL of an image. |
+| classifier | string | Specifies the image, e.g. Thumbnail or Medium. |
```

Comment:
classifier is enum of [Thumbnail, Small, HotDeal, MediumSmall, Medium, MediumLarge, Large]